### PR TITLE
fix(puid/pgid): only set them when running as root, and not explicitly disabled

### PR DESCRIPTION
**Pull request by stavros-k**
**Description**

⚒️ Fixes  # 

**⚙️ Type of change**

- [ ] ⚙️ Feature/App addition
- [x] 🪛 Bugfix
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🔃 Refactor of current code

**🧪 How Has This Been Tested?**

**📃 Notes:**

**✔️ Checklist:**

- [x] ⚖️ My code follows the style guidelines of this project
- [x] 👀 I have performed a self-review of my own code
- [ ] #️⃣ I have commented my code, particularly in hard-to-understand areas
- [ ] 📄 I have made corresponding changes to the documentation
- [x] ⚠️ My changes generate no new warnings
- [x] 🧪 I have added tests to this description that prove my fix is effective or that my feature works
- [x] ⬆️ I increased versions for any altered app according to semantic versioning

**➕ App addition**

If this PR is an app addition please make sure you have done the following.

- [ ] 🪞 I have opened a PR on [truecharts/containers](https://github.com/truecharts/containers) adding the container to TrueCharts mirror repo.
- [ ] 🖼️ I have added an icon in the Chart's root directory called `icon.png`

---

_Please don't blindly check all the boxes. Read them and only check those that apply.
Those checkboxes are there for the reviewer to see what is this all about and
the status of this PR with a quick glance._

### CODE REVIEW

File: charts/common/templates/lib/controller/_container.tpl [hunk 1]
```diff
@@ -50,22 +50,30 @@
   {{- end }}
 
   env:
+   {{- if not .Values.podSecurityContext.runAsUser }}
+    {{- if .Values.security.PUID }}
```
Comment (Ornias1993) [inline charts/common/templates/lib/controller/_container.tpl:54]:
```suggestion
   {{- if and ( not .Values.podSecurityContext.runAsUser) ( .Values.security.PUID ) }}
```

File: charts/common/templates/lib/controller/_container.tpl [hunk 2]
```diff
@@ -50,22 +50,30 @@
...
     - name: UID
       value: {{ tpl ( toYaml .Values.security.PUID ) $ | quote }}
+    {{- end }}
+   {{- end }}
     - name: UMASK
       value: {{ tpl ( toYaml .Values.security.UMASK ) $ | quote }}
     - name: UMASK_SET
       value: {{ tpl ( toYaml .Values.security.UMASK ) $ | quote }}
+   {{- if not .Values.podSecurityContext.runAsGroup }}
+    {{- if .Values.security.PUID }}
```
Comment (Ornias1993) [inline charts/common/templates/lib/controller/_container.tpl:68]:
```suggestion
   {{- if and ( not .Values.podSecurityContext.runAsUser) ( .Values.security.PUID ) }}
```

File: charts/common/templates/lib/controller/_container.tpl [hunk 3]
```diff
@@ -50,22 +50,30 @@
...
   env:
+   {{- if not .Values.podSecurityContext.runAsUser }}
+    {{- if .Values.security.PUID }}
     - name: PUID
       value: {{ tpl ( toYaml .Values.security.PUID ) $ | quote }}
     - name: USER_ID
       value: {{ tpl ( toYaml .Values.security.PUID ) $ | quote }}
     - name: UID
       value: {{ tpl ( toYaml .Values.security.PUID ) $ | quote }}
+    {{- end }}
```
Comment (Ornias1993) [inline charts/common/templates/lib/controller/_container.tpl:61]:
```suggestion
```

**Comment by stavros-k** [comment]
I couldn't find a more elegant way to fix tests.

I tried using `contain` but that checks a list.

**Comment by Ornias1993** [comment]
Tests look solid :)

**Pull request merged**